api_v2/src/notifications/notifications.repository.ts

```diff
@@ -41,7 +44,48 @@ export class NotificationRepository {
     return updatedRequest as ESLChangeShiftRequest;
   }
 
-  async updateNotificationType({
+  async getEslOffsetRequestIdFromNotification(notificationId: number): Promise<number> {
+    const notification = await this.prisma.notification.findUnique({
+      where: { id: notificationId },
+      select: { relatedEntityId: true },
+    });
+
+    if (!notification || !notification.relatedEntityId) {
+      throw new Error('Notification not found or does not have a valid related ESL Offset Request ID.');
+    }
+
+    return notification.relatedEntityId;
+  }
+
+  async updateEslOffsetApproval(requestId: number, isApproved: boolean): Promise<ESLOffset> {
```

Comment:
the updatedOffset condition  is unnecessary since the  Prisma .update() already throws an error if the record isn't found . You can rewrote your code like this

```suggestion
  async updateEslOffsetApproval(requestId: number, isApproved: boolean): Promise<ESLOffset> {

  try {
    return await this.prisma.eslOffset.update({
      where: { id: requestId },
      data: { isLeaderApproved: isApproved },
    });
  } catch (error) {
    throw new Error('Failed to update ESL Offset request.');
  }
}

Reply:
@Mjdelima20  Noted I will refactor